File: frontend/src/services/filePicker.ts
import { api } from "../api/client";

export interface FileEntry {
  relPath: string;
  size: number;
  isText: boolean;
}

export interface ListFilesResponse {
  files: FileEntry[];
  root: string;
}

export interface ReadFileResponse {
  relPath: string;
  uri: string;
  mimeType: string;
  content: string;
  size: number;
}

export interface ReadFileResult {
  relPath: string;
  uri: string;
  mimeType: string;
  content: string | null;
  size: number;
  error: string | null;
}

export interface BatchReadFilesResponse {
  files: ReadFileResult[];
  totalSize: number;
}

export async function listFiles(
  workspaceId: string,
  pattern?: string,
): Promise<ListFilesResponse> {
  const params = new URLSearchParams();
  params.set("workspace_id", workspaceId);
  if (pattern) params.set("pattern", pattern);

  return api.get<ListFilesResponse>(`/file-picker?${params}`);
}

export async function batchReadFiles(
  workspaceId: string,
  paths: string[],
): Promise<BatchReadFilesResponse> {
  return api.post<BatchReadFilesResponse>("/file-picker/batch-read", {
    workspace_id: workspaceId,
    paths,
  });
}

File: crates/agentdash-api/src/dto/file_picker.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct ListFilesQuery {
    pub pattern: Option<String>,
    pub workspace_id: Option<String>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FileEntry {
    pub rel_path: String,
    pub size: u64,
    pub is_text: bool,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ListFilesResponse {
    pub files: Vec<FileEntry>,
    pub root: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ReadFileRequest {
    pub rel_path: String,
    pub workspace_id: Option<String>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ReadFileResponse {
    pub rel_path: String,
    pub uri: String,
    pub mime_type: String,
    pub content: String,
    pub size: u64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct BatchReadFilesRequest {
    pub paths: Vec<String>,
    pub workspace_id: Option<String>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BatchReadFilesResponse {
    pub files: Vec<ReadFileResult>,
    pub total_size: u64,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ReadFileResult {
    pub rel_path: String,
    pub uri: String,
    pub mime_type: String,
    pub content: Option<String>,
    pub size: u64,
    pub error: Option<String>,
}
